docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+to echo back these details. This will include:
+
+* Any response data
+* Information observed from the request such as:
+  * Request headers
+  * Any timeout specified
+  * All request bodies received
+  * Any query parameters observed (only applicable for GET operations such as `IdempotentUnary`)
+
+In all, there are six total endpoints you will need to implement. Below is a brief description of each with helpful pseudocode
+for your implementation.
+
+### Unary
+
+The `Unary` endpoint is a unary operation that accepts a single request of type `UnaryRequest` and returns a single 
+response of type `UnaryResponse`. 
+
+The `UnaryRequest` contains a `response_definition` that tells the server how to respond. This definition contains a 
+`oneof` which specifies whether the server should return valid response data or return an error. Servers should also 
+allow this response definition to be unset. In which case, they should set no response headers or trailers and return 
+no response data. The returned conformance payload should only contain the observed request information.
+
+**Pseudocode**
+
+```text
+read the request
+
+capture any request headers and the actual request body as request info
+
+if a response definition was specified then
+  if response definition specifies valid response data then
+
+    build a conformance payload
+
+    set the request info and response data into the conformance payload
+
+  else
+    build an error with the specified error and set the request info into the error details
+
+  set any response headers or trailers indicated in the response definition
+
+else
+  build a conformance payload
+
+  set the request info into the conformance payload
+
+sleep for any specified response delay
+
+return the response with conformance payload or raise the error, depending on which was specified
+```
+
+For the full documentation on implementing the `Unary` endpoint, click [here][unary].
+
+
+### IdempotentUnary
+
+The `IdempotentUnary` endpoint is also a unary operation. It accepts a request of type  `IdempotentUnaryRequest` and returns
+a single response of type `IdempotentUnaryResponse`. It should be handled in mostly the same way as `Unary`.
+
+The only major difference is that this endpoint should be invoked via an HTTP `GET`. As a result, there is no request body
+so the endpoint should read any query params and set them accordingly in the `connect_get_info` field of the 
+returned `ConformancePayload`. This RPC is the only one that sets the `connect_get_info` field.
+
+For the full documentation on implementing the `IdempotentUnary` endpoint, click [here][idempotentunary].
+
+### Unimplemented
+
+The `Unimplemented` endpoint is also a unary operation, but contrary to the above unary endpoints, the implementation
+of `Unimplemented` should simply return an `unimplemented` error. It is not necessary to echo back any request information or
```

Comment:
Actually, it should not be implemented at all. This tests that the server framework can automatically return the "unimplemented" error for unimplemented methods. Part of the gRPC spec is that adding methods to a service is a safe/compatible change, and that server frameworks should handle this by automatically responding to methods that are missing from the actual server handler implementation. (In connect-go, this actually gets handled by the generated code, in the `Unimplemented...` generated server handler struct, instead of elsewhere in the connect-go runtime code.)